### DISCUSSION TITLE
Lazy rendering of large forms?

### DISCUSSION THREAD

**Discussion by MartinMalinda** [vueform/vueform#272] [Questions]
My usecase for VueForm is to load Airtable fields, convert them to VueForm scheme and pass it to `useVueForm`. It works but if I'm dealing with a large table, generating a form for it does a pretty significant jank or even several seconds of waiting.

I'd be happy if the rendering of the form could be chunked, element by element from top to bottom.

Has anyone experienced with this?

At the moment the only realistic solution is to abandon the stragy of passing schema and instead just do `v-for` and dynamically render the right components myself.

aka

```vue
<Lazy v-for="field in fields">
 <component :is="field.component" v-bind="field.props" />
</Lazy>
```

Or something like that 🤔

**Comment by MartinMalinda** [comment]
I'm trying to keep mutating the schema one element after another but it seems like the component does not dynamically update when schema changes.

```ts
    // recursively construct trimmedSchema by adding fields from newSchema in idleCallback
    const trimmedSchema = {};
    const keys = Object.keys(newSchema);
    console.log({ keys, trimmedSchema });
    const addFormElements = () => {
      const key = keys.shift();
      if (!key) {
        return;
      }

      trimmedSchema[key] = newSchema[key];
      vueform.value = {
        schema: trimmedSchema
      }

      console.log('changing schema', vueform.value.schema);

      if (keys.length) {
        window.requestIdleCallback(addFormElements);
      }
    };
    addFormElements();

    return {
      ...form,
      vueform,
    };
```

**Comment by Acentrix-Jordan** [comment]
+1 this
